tests/epyccel/test_arrays.py

```diff
@@ -4032,6 +3474,20 @@ def test_iterate_slice(language):
     i = randint(2, 10)
     assert f1(i) == f2(i)
 
+#==============================================================================
+# NUMPY SUM
+#==============================================================================
+
+def test_arr_bool_sum(language):
+    f1 = arrays.arr_bool_sum
+    f2 = epyccel(f1, language = language)
+    assert f1() == f2()
```

Comment:
It would be good to check that the return type is as expected:
```
assert isinstance(f1(), type(f2()))
```

Reply:
Done!

Reply:
@EmilyBourne Is this point resolved?

Reply:
It is fixed here, but I don't think there are any tests for the dtype of sums of any other types.

Reply:
By the way, I think it is better if we check exact type equality rather than using `isinstance`. In the latter case `type(f1())` could be a subclass of `type(f2())`, which is not what we want. I recommend instead
```python
assert type(f1()) == type(f2())
```

Reply:
I think it should be:
```
assert type(f1()) is type(f2()) #pylint: disable=unidiomatic-typecheck
```

Reply:
And we should disable the pylint check on the whole test file...

Reply:
The unit tests can be improved in a separate PR